spec/code-snippets/DeclarationReferences.ts

```diff
@@ -0,0 +1,443 @@
+// TSDoc "declaration references" provide a straightforward and unambiguous notation
+// for referencing other API declarations such as classes, member functions, enum values,
+// etc.  It is used with standard TSDoc tags such as {@link} and {@inheritDoc}.
+//
+// The full declaration reference syntax supports declarations that are imported from
+// another NPM package, optionally with an NPM scope, optionally with an explicit
+// import path.
+//
+// For example:
+//
+// /**
+//  * {@link @my-scope/my-package/path1/path2:namespace1.namespace2.MyClass.myMember
+//  * | a complex example}
+//  */
+//
+// The optional components to the left of the ":" simply follow the standard rules of
+// a TypeScript "import" definition, so we don't discuss them further here.
+//
+// TSDoc declaration references are always resolved relative to a specific entry point
+// (NOT relative to the current source file or declaration scope).  Thus, their syntax
```

Comment:
>NOT relative to the current source file or declaration scope [](start = 4, length = 60)

This means that it's effectively impossible to refer to something that isn't exported from the package.